### PR TITLE
Prefix session cookies with app name

### PR DESCRIPTION
**Pull request by sandulat**
Closes: https://github.com/blitz-js/blitz/issues/1153

### What are the changes and their implications?

I've tried to automatically prefix the cookies based on the app's name, using `process.env.npm_package_name`, however, this variable is not available on the front-end.

In my implementation, the session can be prefixed by setting the optional `SESSION_PREFIX` env variable. This way, the prefix is available both on the server and the client.

Example:
```
SESSION_PREFIX=myAppName
```

### Checklist

- [ ] Tests added for changes
- [ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes

### CODE REVIEW

File: packages/core/src/supertokens.ts [hunk 1]
```diff
@@ -2,18 +2,23 @@ import {useState} from "react"
 import BadBehavior from "bad-behavior"
 import {useIsomorphicLayoutEffect} from "./utils/hooks"
 import {queryCache} from "react-query"
+import getConfig from "next/config"
+
+const {publicRuntimeConfig = {}} = getConfig() || {}
```
Comment (sandulat) [inline packages/core/src/supertokens.ts:7]: Next's `getConfig` will return the config on the client only.

File: packages/core/src/supertokens.ts [hunk 2]
```diff
@@ -2,18 +2,23 @@ import {useState} from "react"
...
+
+const sessionPrefix = publicRuntimeConfig.sessionPrefix || process.env.SESSION_PREFIX || "s"
```
Comment (sandulat) [inline packages/core/src/supertokens.ts:9]: `process.env.SESSION_PREFIX` will be available on the server only.

**Comment by flybayer** [comment]
Thanks @sandulat! Let's find a way to make it work automatically by default.

Perhaps in withBlitz we can do this? Anything in `env` will be available in the frontend via `process.env.sessionPrefix`

```
env: {
  sessionPrefix: process.env.npm_package_name || process.env.SESSION_PREFIX
}
```

But we'd also need to merge in `env` from the user's blitz.config.js

**Comment by sandulat** [comment]
Great idea @flybayer! I've implemented it. The only thing is that when the app's name in `package.json` equals for example `@examples/auth`, the session is being set, however, the authentication doesn't effectively happen. It feels like the server can't match the cookies. When the name is just `auth`, everything works perfectly. It looks to me that one of the `@` `/` symbols are somewhere stripped, but I don't really understand where, and which is the pattern.

**Comment by flybayer** [comment]
Cool! I think we should replace any non-url safe characters with and underscore. So `@examples/auth` would turn to `_examples_auth`.

Also, I had a new concern about the process.env thing. Can you try setting `isInternalBlitzDevelopment` to `false` in `withBlitz` and see if it still works? The way the env values are inlined via webpack is a bit goofy, so I want to make sure it works without the transpile modules plugin we have there in the monorepo.

**Comment by sandulat** [comment]
@flybayer I've set isInternalBlitzDevelopment to false, and everything seems to work just fine. I've also pushed the replacement of non-url safe characters in the prefix.

**Comment by flybayer** [comment]
@sandulat Sweet, thanks! I'll go ahead and merge, but please also open a PR to the docs site to document this new functionality.

**Pull request merged**